File: juzgo/juzgo/custom/js/salary_slip.js
frappe.ui.form.on('Salary Slip', {
    refresh: function(frm) {
        frappe.call({
            method: 'juzgo.juzgo.custom.py.salary_slip.get_last_from_date',
            args: {
                'employee': frm.doc.employee,
                'salary_structure': frm.doc.salary_structure,
                'posting_date': frm.doc.posting_date
            },
            callback: function(response) {
                if (response && response.message) {
                    var lastRecord = response.message;
                    frm.set_value('salary_amount', lastRecord.base);
                }
            }
        });
    },
    custom__monthly_attendance_details: function(frm){
        let dateStr = frm.doc.posting_date;
        let dateParts = dateStr.split("-");
        let monthNumber = parseInt(dateParts[1], 10);
        let year = new Date(dateStr).getFullYear();
        frappe.set_route('query-report', 'Monthly Attendance Details', {month:monthNumber, year:year, employee: frm.doc.employee});
    },
});

File: juzgo/juzgo/custom/py/salary_slip.py

import frappe

@frappe.whitelist()
def get_last_from_date(employee=None, salary_structure=None, posting_date=None):
    last_salary_assignment = frappe.get_list(
        'Salary Structure Assignment',
        filters={
            'employee': employee,
            'salary_structure': salary_structure,
            'from_date': ('<', posting_date)
        },
        fields=['base', 'from_date'],
        order_by='from_date DESC',
        limit=1
    )

    return last_salary_assignment[0] if last_salary_assignment else None
